### ISSUE TITLE
download-antismash-databases does not use the --database-dir argument in the "Pre-building all databases" step

### ISSUE THREAD

**Issue by mmp3** [antismash/antismash#370]
**Describe the bug**

For `antismash-6.0.0`, installed via "full manual install", the `download-antismash-databases` command ignores the `--database-dir` argument in the "Pre-building all databases" step.

Specifically,  the command:

```
$ download-antismash-databases  --database-dir /data1/antismash_db
```

Gives output:

```
Creating checksum of Pfam-A.hmm
PFAM file present and ok for version 32.0
Downloading Resfam database
Downloading Resfams.hmm.gz: 100.00% downloaded.
Creating checksum of Resfams.hmm.gz
Extraction of Resfams.hmm.gz finished successfully.
Ensuring all cutoffs are present
Creating checksum of TIGRFam.hmm
TIGRFam database present and checked
Creating checksum of proteins.fasta
ClusterBlast fasta file present and checked
Creating checksum of proteins.fasta
ClusterCompare mibig FASTA file present and checked
Pre-building all databases...
Error encountered while preparing module data: [Errno 13] Permission denied: '/usr/local/lib/python3.8/dist-packages/antismash/detection/hmm_detection/data/bgc_seeds.hmm'
```

All of the databases have been correctly downloaded to `/data1/antismash_db`, as instructed in the `--database-dir` argument, but the final "pre-build" step is trying to write data to the Python 3.8 packages directory, which it should not be doing.

**Expected behavior**
"Pre-building all databases" step should write any additional database data to the location specified by the `--database-dir` argument to `download-antismash-databases`.

**To Reproduce**

```
$ download-antismash-databases  --database-dir /data1/antismash_db
```

**System:**
```
$ lsb_release  -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
```

**Comment by SJShaw** [comment]
The database directory option is for the large databases that aren't shipped with antiSMASH source, Pfam, clusterblast, etc. Those databases downloaded will respect the `--database-dir` option.

However, there's a number of small files that are created on first run (or with `--check-prereqs`), most of which depend on the versions of the dependencies and are saved to disk to save time on every subsequent run. This includes the normal `*.pyc` files that python will generate for the same reason. Among these are the `bgc_seeds.hmm` that needs to be pressed with your installed `hmmpress`, classifiers for your version of `scikit-learn` are built and saved, and so on.

**Comment by mmp3** [comment]
As proof that `download-antismash-databases` indeed ignores the `--database-dir` argument during the checking step, if I manually copy all files from the `--databae-dir` to the `site-packages/antismash/databases/` directory, then it works.

Specifically:

```
# copy everything from `--database-dir` to `--site-packages`:
$ rsync -rtvh /data1/antismash_db/ /home/ubuntu/packages/python_3.8_venv_20210618/lib/python3.8/site-packages/antismash/databases/

# try again
$ download-antismash-databases
```

gives output:


```
Creating checksum of Pfam-A.hmm
PFAM file present and ok for version 32.0
Downloading Resfam database
Downloading Resfams.hmm.gz: 100.00% downloaded.
Creating checksum of Resfams.hmm.gz
Extraction of Resfams.hmm.gz finished successfully.
Ensuring all cutoffs are present
Creating checksum of TIGRFam.hmm
TIGRFam database present and checked
Creating checksum of proteins.fasta
ClusterBlast fasta file present and checked
Creating checksum of proteins.fasta
ClusterCompare mibig FASTA file present and checked
Pre-building all databases...
done.
```

**Comment by LuisFF** [comment]
I'm not sure this issue is fixed in version `6.0.0` given that `download-antismash-databases` still tries to write to `dist-packages`
```
Downloading clusterblast_20201113.tar.xz: 99.56% downloaded.Error encountered while preparing module data: [Errno 2] No such file or directory: '<PYTHON_HOME>/dist-packages/antismash/databases/clusterblast/proteins.fasta
```

Is there a way to split the process of updating  "large databases that aren't shipped with antiSMASH source" from updating "small files that are created on first run" ? Also, updating package data files "on first run" is not ideal when running on stateless systems.

**Comment by SJShaw** [comment]
We don't re-release the same version numbers, so it won't be fixed in 6.0.0. If you install 6.0.1 or later, it will include the fix.

Unless I'm misinterpreting something in the rest of your comment, the it's already addressed [above](https://github.com/antismash/antismash/issues/370#issuecomment-864916121).

**Comment by LuisFF** [comment]
That's reasonable. I must have misunderstood the tagging system, in particular tag 6-0-stable. It makes sense now. Thanks.
